### PR TITLE
Enable PYPI upload workflow without Windows packages

### PR DESCRIPTION
**Pull request by dudoslav**
This PR implements PyPi upload for linux and macosx platforms since windows pipeline still fails. Source distribution is available as well.

The previous PR that fails and is postponed is here: https://github.com/TileDB-Inc/TileDB-Vector-Search/pull/207

Edit: Recreate PR from this repo instead of fork so that all tests can pass

### CODE REVIEW

**Comment by ihnorton** [comment]
@jparismorgan please give this a once-over and merged if ok w/ you.

File: .github/workflows/build-wheels.yml
Comment (jparismorgan) [inline .github/workflows/build-wheels.yml:47]: Was this a merge conflict? I believe we need to keep this as it was.

**Review by jparismorgan — APPROVED** [review]
Approving with the assumption that the `pip install .` change was an accident and is reverted before merge. If that's the case then looks good to me.

File: .github/workflows/build-wheels.yml
Comment (dudoslav) [reply .github/workflows/build-wheels.yml:47]: https://github.com/dudoslav/TileDB-Vector-Search/actions/runs/7916707804/job/21611139539#step:6:7

but it seems that it works

Comment (dudoslav) [reply .github/workflows/build-wheels.yml:47]: I think that that pipeline would not work on release

Comment (dudoslav) [reply .github/workflows/build-wheels.yml]: Returned

**Review by jparismorgan — APPROVED** [review]

**Pull request merged**
